cou/apps/auxiliary.py

```diff
@@ -93,3 +87,41 @@ def channel_codename(self) -> OpenStackRelease:
                 f"for channel: '{self.channel}'"
             )
         )
+
+    @property
+    def channel(self) -> str:
+        """Get charm channel of the application.
+
+        :return: Charm channel. E.g: 3.8/stable
+        :rtype: str
+        """
+        return self._channel
+
+    @channel.setter
+    def channel(self, charm_channel: str) -> None:
+        """Set charm channel of the application.
+
+        When application comes from charm store, the channel won't be track related.
+        If the application is subordinate, we can't check the tracks because the OpenStack
+        release is based on the channel itself.
+        :param charm_channel: Charm channel. E.g: 3.8/stable
+        :type charm_channel: str
+        :raises ValueError: Exception raised when cannot find a channel track
+            based on the charm name, series and current OpenStack codename.
+        """
+        if self.is_from_charm_store or self.is_subordinate:
+            self._channel = charm_channel
+        else:
+            tracks = OPENSTACK_TO_TRACK_MAPPING.get(
+                (self.charm, self.series, self.current_os_release.codename), []
+            )
+            track_from_channel = charm_channel.split("/")[0]
+            if track_from_channel not in tracks:
+                raise ValueError(
+                    (
+                        f"'{self.name}' cannot find a channel track for charm: '{charm_channel}' "
+                        f"series: {self.series} and OpenStack "
+                        f"release: {self.current_os_release.codename}"
+                    )
+                )
+            self._channel = charm_channel
```

Comment:
not-blocking: same as in previous one, I think it's ok to use return here
```suggestion
        if self.is_from_charm_store or self.is_subordinate:
            self._channel = charm_channel
            return

        tracks = OPENSTACK_TO_TRACK_MAPPING.get(
            (self.charm, self.series, self.current_os_release.codename), []
        )
        track_from_channel = charm_channel.split("/")[0]
        if track_from_channel not in tracks:
            raise ValueError(
                (
                    f"'{self.name}' cannot find a channel track for charm: '{charm_channel}' "
                    f"series: {self.series} and OpenStack "
                    f"release: {self.current_os_release.codename}"
                )
            )
        self._channel = charm_channel
```